lib/optimizer/src/Transformer/ServerSideRendering.php

```diff
@@ -620,4 +699,260 @@ private function isAmpExperimentUsed(DOMElement $element)
         // Otherwise, used.
         return true;
     }
+
+    /**
+     * Adapt blocking attributes so that they allow for boilerplate removal.
+     *
+     * Blocking attributes that need special attention are `sizes`, `heights` and `media`.
+     *
+     * @see https://github.com/ampproject/amp-wp/issues/4439
+     *
+     * @param Document        $document   DOM document to apply the transformations to.
+     * @param DOMElement      $ampElement Element to adapt.
+     * @param ErrorCollection $errors     Collection of errors that are collected during transformation.
+     * @return bool Whether boilerplate can be removed.
+     */
+    private function adaptBlockingAttributes(Document $document, DOMElement $ampElement, ErrorCollection $errors)
+    {
+        $attributes = $ampElement->attributes;
+
+        if ($attributes === null) {
+            return true;
+        }
+
+        $customCss          = '';
+        $attributesToRemove = [];
+
+        foreach ($attributes as $attribute) {
+            /**
+             * Attribute to check.
+             *
+             * @var DOMAttr $attribute
+             */
+            $normalizedAttributeName = strtolower($attribute->name);
+            if (array_key_exists($normalizedAttributeName, self::EXTRACTION_METHOD_MAP)) {
+                if (
+                    $normalizedAttributeName === Attribute::SIZES
+                    && $ampElement->hasAttribute(Attribute::DISABLE_INLINE_WIDTH)
+                ) {
+                    // Don't remove sizes when disable-inline-width is set.
+                    // @see https://github.com/ampproject/amphtml/pull/27083
+                    continue;
+                }
+
+                $method = self::EXTRACTION_METHOD_MAP[$normalizedAttributeName];
+
+                try {
+                    $customCss            .= $this->$method($document, $ampElement, $attribute->value);
+                    $attributesToRemove[] = $attribute->name;
```

Comment:
The equals signs aren't lined up. Is that intended?

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/4482/commits/facc01606bab638a58ff32a991becc6686ceb94a